lib/publiccloud/provider.pm

```diff
@@ -269,23 +293,40 @@ sub terraform_apply {
     my $image                = $self->get_image_id();
     my $ssh_private_key_file = '/root/.ssh/id_rsa';
     my $name                 = get_var('PUBLIC_CLOUD_RESOURCE_NAME', 'openqa-vm');
+    my $cloud_name           = $self->get_sap_cloud_name;
 
     record_info('WARNING', 'Terraform apply has been run previously.') if ($self->terraform_applied);
 
-    assert_script_run('cd ' . TERRAFORM_DIR);
+    if (check_var('SLES4SAP_MODE', 'publiccloud')) {
+        assert_script_run('cd ' . TERRAFORM_DIR . "/$cloud_name");
```

Comment:
as far as I remember we don't need this because usually each worker working with some specific PC provider , why it is not working for you ?

Reply:
Files are in separate directories for each worker, so we need to go inside the corresponding directory.

Reply:
what you mean "separate directories for each worker" ? The worker which run the job for amazon will use terraform to start **only** amazon , for google cloud or azure there will different worker running different job so you can easily put profiles directly into TERRAFORM_DIR and they will never mixed because there is no case when same job doing something for amazon and azure in one shot

Reply:
Yes, but I clone the Git repository as-is in `TERRAFOM_DIR` and I'm not changing the structure.

If you check [here](https://github.com/SUSE/ha-sap-terraform-deployments) you will see different directories for each cloud provider. The `ha-sap-terraform-deployments` tool has been made to be able to run tests for each providers from the same server.